### PR TITLE
Initial try for connection restart

### PR DESCRIPTION
**Pull request by skyline75489**
## Summary of the Pull Request

This targets #3726 & #4379

Surprisingly, it works.

 
## PR Checklist
* [ ] Closes #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

### CODE REVIEW

File: src/cascadia/TerminalConnection/ConptyConnection.cpp [hunk 1]
```diff
@@ -436,6 +436,43 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
         _hPC.reset();
     }
 
+
+    void ConptyConnection::Restart()
+    try
+    {
+        THROW_IF_FAILED(_LaunchAttachedClient());
```
Comment (lhecker) [inline src/cascadia/TerminalConnection/ConptyConnection.cpp:443]: Don't we have to kill the client before creating a new one?

**Review by DHowett — CHANGES_REQUESTED** [review]
So! In this case, I think it would be better for us to _delete the old connection object and create a new one._ That will simplify the Connection interface (not all connections can be "restarted"), and you will not have to ever handle a "backwards" state transition.

`ITerminalConnection` should represent the things that any connection could do, even after we get a plug-in model. If we can implement restart on top of "stop, delete, create new, start" we can free plugin-in authors from having to think about tracking any of their local state across multiple different sessions, etc.

What do you think about that?

**Comment by zadjii-msft** [comment]
FWIW I agree with Dustin on this one. I think stashing the `ValueSet` for the connection settings should be an easy enough way of keeping all the information relevant to starting a new clone of the connection.

**Comment by skyline75489** [comment]
Yeah you guys are right. This PR is just a demonstration.

**Pull request closed without merging**